### PR TITLE
Add descriptor set layouts to frame looping

### PR DESCRIPTION
**Pull request by purvisa-at-google-com**

### CODE REVIEW

File: gapis/api/vulkan/frame_loop.go [hunk 1]
```diff
@@ -643,3 +708,21 @@ func (f *frameLoop) copyImage(ctx context.Context, srcImg, dstImg ImageObjectʳ,
 
 	return nil
 }
+
+func (f *frameLoop) resetDescriptorSetLayouts(ctx context.Context, stateBuilder *stateBuilder) error {
+
+	// For every DescriptorSetLayout that was created during the loop...
+	for created, _ := range f.descriptorSetLayoutCreated {
```
Comment (RenfengLiu) [inline gapis/api/vulkan/frame_loop.go]: Maybe `for created := range f.descriptorSetLayoutCreated {`

File: gapis/api/vulkan/frame_loop.go [hunk 2]
```diff
@@ -297,175 +494,235 @@ func (f *frameLoop) detectChangedResource(ctx context.Context, startState *api.G
...
+		// Write the command to delete the created object
+		dsl := GetState(f.loopEndState).descriptorSetLayouts.Get(created)
+		stateBuilder.write(stateBuilder.cb.VkDestroyDescriptorSetLayout(dsl.Device(), dsl.VulkanHandle(), memory.Nullptr))
+	}
+
+	// For every DescriptorSetLayout that was destroyed during the loop...
+	for destroyed, _ := range f.descriptorSetLayoutDestroyed {
```
Comment (RenfengLiu) [inline gapis/api/vulkan/frame_loop.go]: The same as above

File: gapis/api/vulkan/frame_loop.go [hunk 3]
```diff
@@ -365,10 +387,44 @@ func (f *frameLoop) detectChangedResource(ctx context.Context, startState *api.G
...
+			vkCmd := cmd.(*VkDestroyDescriptorSetLayout)
+			descriptorSetLayout := vkCmd.DescriptorSetLayout()
+			log.D(ctx, "DescriptorSetLayout %v created", descriptorSetLayout)
+			if _, ok := f.descriptorSetLayoutCreated[descriptorSetLayout]; ok {
+				delete(f.descriptorSetLayoutCreated, descriptorSetLayout)
+			} else {
+				f.descriptorSetLayoutDestroyed[descriptorSetLayout] = true
+			}
+
+		// DescriptorPool(s)
```
Comment (AWoloszyn) [inline gapis/api/vulkan/frame_loop.go]: I would prefer not checking in commented out / empty code.

File: gapis/api/vulkan/frame_loop.go (hunk 1 above)
Comment (AWoloszyn) [reply gapis/api/vulkan/frame_loop.go]: Good catch @RenfengLiu. You can just drop the _.

**Comment by purvisa-at-google-com** [comment]
We good to submit this one?

**Pull request merged**